packages/react-components/src/components/VideoGallery/LargeGalleryLayout.tsx

```diff
@@ -0,0 +1,191 @@
+
+  const overflowGalleryTiles = overflowGalleryParticipants.map((p, i) => {
+    return onRenderRemoteParticipant(
+      p,
+      maxRemoteVideoStreams && maxRemoteVideoStreams >= 0
+        ? p.videoStream?.isAvailable && indexesToRender.includes(i) && activeVideoStreams++ < maxRemoteVideoStreams
+        : p.videoStream?.isAvailable
+    );
+  });
+
+  if (localVideoComponent) {
+    gridTiles.push(localVideoComponent);
+  }
+
+  const overflowGallery = useMemo(() => {
+    if (overflowGalleryTiles.length === 0) {
+      return null;
+    }
+    return (
+      <OverflowGallery
+        isNarrow={isNarrow}
+        /* @conditional-compile-remove(gallery-layouts) */
+        isShort={isShort}
+        shouldFloatLocalVideo={false}
+        overflowGalleryElements={overflowGalleryTiles}
+        horizontalGalleryStyles={styles?.horizontalGallery}
+        /* @conditional-compile-remove(vertical-gallery) */
+        verticalGalleryStyles={styles?.verticalGallery}
+        /* @conditional-compile-remove(vertical-gallery) */
+        overflowGalleryPosition={overflowGalleryPosition}
+        onFetchTilesToRender={setIndexesToRender}
+        onChildrenPerPageChange={(n: number) => {
+          childrenPerPage.current = n;
+        }}
+      />
+    );
+  }, [
+    isNarrow,
+    /* @conditional-compile-remove(gallery-layouts) */ isShort,
+    overflowGalleryTiles,
+    styles?.horizontalGallery,
+    /* @conditional-compile-remove(vertical-gallery) */ overflowGalleryPosition,
+    setIndexesToRender,
+    /* @conditional-compile-remove(vertical-gallery) */ styles?.verticalGallery
+  ]);
+
+  return (
+    <Stack
+      /* @conditional-compile-remove(vertical-gallery) */
+      horizontal={overflowGalleryPosition === 'VerticalRight'}
+      styles={rootLayoutStyle}
+      tokens={videoGalleryLayoutGap}
+    >
+      {
+        /* @conditional-compile-remove(gallery-layouts) */ props.overflowGalleryPosition === 'HorizontalTop' ? (
+          overflowGallery
+        ) : (
+          <></>
+        )
+      }
+      {screenShareComponent ? (
+        screenShareComponent
+      ) : (
+        <GridLayout key="grid-layout" styles={styles?.gridLayout}>
+          {gridTiles}
+        </GridLayout>
+      )}
+      {overflowGalleryTrampoline(
+        overflowGallery,
+        /* @conditional-compile-remove(gallery-layouts) */ props.overflowGalleryPosition
+      )}
+    </Stack>
+  );
+};
+
+const overflowGalleryTrampoline = (
+  gallery: JSX.Element | null,
+  galleryPosition?: 'HorizontalBottom' | 'VerticalRight' | 'HorizontalTop'
+): JSX.Element | null => {
+  /* @conditional-compile-remove(gallery-layouts) */
+  return galleryPosition !== 'HorizontalTop' ? gallery : <></>;
+  return gallery;
+};
+
+/* @conditional-compile-remove(gallery-layouts) */
+const calculateMaxTilesInLargeGrid = (parentWidth: number, parentHeight: number): number => {
+  const xAxisTiles = Math.floor(parentWidth / (VERTICAL_GALLERY_TILE_SIZE_REM.width * REM_TO_PIXEL));
```

Comment:
Why are we using vertical gallery tile size here?

Reply:
these tiles are smaller and used as a good benchmark for what a 'smallest' tile could be, think I should make a new constant?

Reply:
I see. That's alright. Maybe we can refactor the name of VERTICAL_GALLERY_TILE_SIZE_REM to something like SMALLEST_TILE_SIZE_REM. But for a refactor PR after once everything is in.